File: backend/app/schemas/emission_factor.py
"""Pydantic schemas for the Emission Factor resource.

Matches docs/api-contract.md — Emission Factors section.
Read-only in the MVP: seeded via app/seed.py, never created through the API.
"""

from datetime import date
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel, ConfigDict


class EmissionFactorResponse(BaseModel):
    """GET /emission-factors response item."""
    model_config = ConfigDict(from_attributes=True)

    id: int
    source_type: str
    region: str
    factor_value: Decimal
    unit: str
    valid_from: date
    valid_to: Optional[date]
    source_reference: str

File: frontend/src/types/emissionFactor.ts
/** Mirrors docs/api-contract.md — Emission Factors section. Read-only; seeded server-side. */

import type { SourceType } from "./emissionSource";

export interface EmissionFactor {
  id: number;
  source_type: SourceType;
  region: string;
  /** Decimal string, e.g. "0.708200" — never parse and re-render as a float. */
  factor_value: string;
  unit: string;
  valid_from: string;
  valid_to: string | null;
  source_reference: string;
}
